include/ginkgo/core/base/math.hpp

```diff
@@ -388,7 +388,13 @@ GKO_INLINE __host__ constexpr T zero(const T &)
  * @return the multiplicative identity for T
  */
 template <typename T>
-GKO_INLINE __host__ constexpr T one()
+GKO_INLINE __device__ constexpr xstd::enable_if_t<is_complex<T>(), T> one()
+{
+    return T(one<remove_complex<T>>());
+}
+
+template <typename T>
+GKO_INLINE __device__ constexpr xstd::enable_if_t<!is_complex<T>(), T> one()
```

Comment:
Should this not also be `__host__` (since the original was host)?
```suggestion
GKO_INLINE __host__ constexpr xstd::enable_if_t<!is_complex<T>(), T> one()
```